packages/victory-core/src/victory-util/data.js

```diff
@@ -91,6 +92,67 @@ function cleanData(dataset, props) {
   });
 }
 
+// This method will remove data points that fall outside of the desired domain (non-continuous charts only)
+function formatDataFromDomain(dataset, props) {
+  const exists = (val) => val !== undefined;
+
+  // domain for checking data points that are outside of the bounds
+  // symbol to determine chart types with a single y value point
+  // interpolation for determining chart types with continuous data points that shouldn't be altered
+  const { domain, interpolation } = props;
+
+  if (interpolation) return dataset;
+
+  const minDomainX =
+    !domain || !domain.x ? Domain.getMinFromProps(props, "x") : Collection.getMinValue(domain.x);
+  const maxDomainX =
+    !domain || !domain.x ? Domain.getMaxFromProps(props, "x") : Collection.getMaxValue(domain.x);
+  const minDomainY =
+    !domain || !domain.y ? Domain.getMinFromProps(props, "y") : Collection.getMinValue(domain.y);
+  const maxDomainY =
+    !domain || !domain.y ? Domain.getMaxFromProps(props, "y") : Collection.getMaxValue(domain.y);
+
+  if (!exists(minDomainX) && !exists(maxDomainX) && !exists(minDomainY) && !exists(maxDomainY))
+    return dataset;
+
+  // eslint-disable-next-line complexity
+  return dataset.map((datum) => {
+    let { _x, _y, _y0, _y1 } = datum;
+
+    const baseline = _y0;
+    const value = _y || _y1;
+
+    // don't alter data points if _y is an array of values
+    if (Array.isArray(value)) return datum;
+
+    // single x point less than min domain
+    if (exists(_x) && _x < minDomainX) _x = null;
```

Comment:
That's a lot of if statements. I think this can be made a lot more succinct by 1) first checking if the value exists. If it doesn't then I guess it's already null, and we can return the datum with no further checks. 2) check if the baseline exists and nest the rests of our checks so we don't have to check every time.

Reply:
I added some curried functions that allowed me simplify the checks quite a bit.  Also quitting early when value doesn't exist.